app/controllers/my_facilities_controller.rb

```diff
@@ -10,6 +10,7 @@ class MyFacilitiesController < AdminController
   PERIODS_TO_DISPLAY = {quarter: 3, month: 3, day: 14}.freeze
 
   around_action :set_time_zone
+  before_action :set_period, except: [:index]
```

Comment:
Is this filter really just for the blood_pressure_control action here? If so, it should have an `only` declaration for that instead of the `except`